CMakeLists.txt

```diff
@@ -230,7 +230,7 @@ OCV_OPTION(WITH_CUDA           "Include NVidia Cuda Runtime support"
 OCV_OPTION(WITH_CUFFT          "Include NVidia Cuda Fast Fourier Transform (FFT) library support"            ON  IF (NOT IOS AND NOT WINRT) )
 OCV_OPTION(WITH_CUBLAS         "Include NVidia Cuda Basic Linear Algebra Subprograms (BLAS) library support" ON IF (NOT IOS AND NOT WINRT) )
 OCV_OPTION(WITH_NVCUVID        "Include NVidia Video Decoding library support"                               ON IF (NOT IOS AND NOT APPLE) )
-OCV_OPTION(WITH_EIGEN          "Include Eigen2/Eigen3 support"               (NOT CV_DISABLE_OPTIMIZATION)       IF (NOT WINRT AND NOT CMAKE_CROSSCOMPILING) )
+OCV_OPTION(WITH_EIGEN          "Include Eigen2/Eigen3 support"               (NOT CV_DISABLE_OPTIMIZATION)       IF (NOT WINRT) )
```

Comment:
It is fine to allow Eigen for cross-compilation.
But lets continue enabling it explicitly via `-DWITH_EIGEN=ON`.
```
(NOT CV_DISABLE_OPTIMIZATION AND NOT CMAKE_CROSSCOMPILING)
```

Reply:
So why does `WITH_EIGEN` defaults to `OFF` when cross-compiling?

Reply:
Because of include directory.

What is going with cross-compilation if you add `/usr/include` (where Eigen is located on host system) into the list of include directories?

Reply:
This is the point of this PR. The host version of Eigen will not be used during cross-compilation anymore.

Reply:
I mean that result of compilation is unpredictable if you add **host** /usr/include (or /usr/local/include) into compiler list of include directories.

This dangerous behavior should be disabled by default.

Reply:
Yes, that's why this PR removed unconditional `/usr/include` and `/usr/local/include` Eigen search prefixes, which are set by CMake anyway. But not during cross-compilation.

The question is now why do only exlude  Eigen (and presumably few other libraries) during cross-compliation, but not other? This is inconsistent.